packages/pigeon/README.md

```diff
@@ -91,8 +91,10 @@ Nested datatypes are supported, too.
 
 ### Asynchronous Handlers
 
-By default Pigeon will generate synchronous handlers for messages.  If you want
-to be able to respond to a message asynchronously you can use the `@async`
+By default Pigeon will generate synchronous handlers for messages. 
+The API can return data synchronously but the request function is always asynchronous.
+
+If you want to be able to respond to a message asynchronously you can use the `@async`
```

Comment:
How about:

By default Pigeon will generate synchronous handlers for messages (i.e. the host code for `@HostApi's` and the Flutter code for `@FlutterApi's`).  Using the `@async` annotation on a method in the Pigeon file makes the generated methods respond asynchronously instead.

Reply:
I try to clarifiy something else that does not have to be obvious to people that see the package for the first time.
That is that the other side will be always async no matter what you do. First time I just saw info that it is synchronous and from the context I did not know it is only one sided synchronous code. I understand that from perspective of a person that is doing this often it might look different and there might be obvious reasons for which it wouldnt make sense that it would be both sides sync. But thats what I thought combined with the info that it is also faster. I can rephrase it a bit more so its obvious from your technical point of view and maybe the more naive point of view.

Reply:
I think this is still a bit confusing because "request function" isn't a term we use.  I think there is some confusion in vocabulary between the generated function the client calls and the handler that gets invoked as a result.

I'd refer to the former as "generated methods" and "handlers".  So the generated methods are always asynchronous, but the handlers are synchronous by default but can be asynchronous with the `@async` annotation.  I don't think this clarifies.

Reply:
suggestion:

By default Pigeon will generate synchronous handlers for messages and asynchronous methods.  If you want a handler to be able to respond to a message asynchronously you can use the `@async` annotation as of version 0.1.20.